In [1]:
import numpy as np
import pandas as pd

text = pd.read_csv('F:/Titanic/train_chinese.csv')
text.head(3)

乘客ID  是否幸存  仓位等级                                                 姓名  \
0     1     0     3                            Braund, Mr. Owen Harris   
1     2     1     1  Cumings, Mrs. John Bradley (Florence Briggs Th...   
2     3     1     3                             Heikkinen, Miss. Laina   

       性别    年龄  兄弟姐妹个数  父母子女个数              船票信息       票价   客舱 登船港口  
0    male  22.0       1       0         A/5 21171   7.2500  NaN    S  
1  female  38.0       1       0          PC 17599  71.2833  C85    C  
2  female  26.0       0       0  STON/O2. 3101282   7.9250  NaN    S

利用Pandas对示例数据进行排序，要求升序

In [7]:
#先创造一个示例dataframe
# 具体请看《利用Python进行数据分析》第五章 排序和排名 部分

#自己构建一个都为数字的DataFrame数据
frame = pd.DataFrame(np.arange(8).reshape((2,4)),
                        index = ['2','1'],
                        columns = ['a','b','c','d'])             #这些都是DataFrame函数传进去的参数，写成一行再展开看的会比较清楚
frame

a  b  c  d
2  0  1  2  3
1  4  5  6  7

In [8]:
#对上面的frame进行升降序
frame = pd.DataFrame(np.arange(8).reshape((2,4)),
                        index = ['2','1'],
                        columns = ['a','b','c','d']) 
frame.sort_values(by='c', ascending=False)         #通过取c让这一列降序排序，于是让每列直接都降序


a  b  c  d
1  4  5  6  7
2  0  1  2  3

In [10]:
frame = pd.DataFrame(np.arange(8).reshape((2,4)),
                        index = ['2','1'],
                        columns = ['a','b','c','d']) 
# 让行索引升序排序           (产生和上面一样的结果)
frame.sort_index()    #列又名axis=0

a  b  c  d
1  4  5  6  7
2  0  1  2  3

In [12]:
frame = pd.DataFrame(np.arange(8).reshape((2,4)),
                        index = ['2','1'],
                        columns = ['a','b','c','d']) 
# 让列索引降序排序   
frame.sort_index(axis=1, ascending = False)

d  c  b  a
2  3  2  1  0
1  7  6  5  4

利用Pandas进行算术计算，计算两个DataFrame数据相加结果

In [14]:
# 具体请看《利用Python进行数据分析》第五章 算术运算与数据对齐 部分

#建立一个例子
frame1_a = pd.DataFrame(np.arange(9.).reshape(3, 3),
                     columns=['a', 'b', 'c'],
                     index=['one', 'two', 'three'])
frame1_b = pd.DataFrame(np.arange(12.).reshape(4, 3),
                     columns=['a', 'e', 'c'],
                     index=['first', 'one', 'two', 'second'])
frame1_a


a    b    c
one    0.0  1.0  2.0
two    3.0  4.0  5.0
three  6.0  7.0  8.0

In [15]:
frame1_a = pd.DataFrame(np.arange(9.).reshape(3, 3),
                     columns=['a', 'b', 'c'],
                     index=['one', 'two', 'three'])
frame1_b = pd.DataFrame(np.arange(12.).reshape(4, 3),
                     columns=['a', 'e', 'c'],
                     index=['first', 'one', 'two', 'second'])
frame1_b

a     e     c
first   0.0   1.0   2.0
one     3.0   4.0   5.0
two     6.0   7.0   8.0
second  9.0  10.0  11.0

In [16]:
frame1_a = pd.DataFrame(np.arange(9.).reshape(3, 3),
                     columns=['a', 'b', 'c'],
                     index=['one', 'two', 'three'])
frame1_b = pd.DataFrame(np.arange(12.).reshape(4, 3),
                     columns=['a', 'e', 'c'],
                     index=['first', 'one', 'two', 'second'])
frame1_a + frame1_b                     

a   b     c   e
first   NaN NaN   NaN NaN
one     3.0 NaN   7.0 NaN
second  NaN NaN   NaN NaN
three   NaN NaN   NaN NaN
two     9.0 NaN  13.0 NaN

如上图所示，匹配的数据会相加，其他没有对应的行列就会显示NaN

通过泰坦尼克号数据如何计算出在船上最大的家族有多少人？

In [17]:
import numpy as np
import pandas as pd

text = pd.read_csv('F:/Titanic/train_chinese.csv')

'''还是用之前导入的chinese_train.csv如果我们想看看在船上，最大的家族有多少人（‘兄弟姐妹个数’+‘父母子女个数’），我们该怎么做呢？'''

max(text['兄弟姐妹个数'] + text['父母子女个数'])

10

In [10]:
import numpy as np
import pandas as pd

text = pd.read_csv('F:/Titanic/train_chinese.csv')

text[(text['兄弟姐妹个数'] + text['父母子女个数'])==10].head(30)

乘客ID  是否幸存  仓位等级                                 姓名      性别  年龄  兄弟姐妹个数  \
159   160     0     3         Sage, Master. Thomas Henry    male NaN       8   
180   181     0     3       Sage, Miss. Constance Gladys  female NaN       8   
201   202     0     3                Sage, Mr. Frederick    male NaN       8   
324   325     0     3           Sage, Mr. George John Jr    male NaN       8   
792   793     0     3            Sage, Miss. Stella Anna  female NaN       8   
846   847     0     3           Sage, Mr. Douglas Bullen    male NaN       8   
863   864     0     3  Sage, Miss. Dorothy Edith "Dolly"  female NaN       8   

     父母子女个数      船票信息     票价   客舱 登船港口  
159       2  CA. 2343  69.55  NaN    S  
180       2  CA. 2343  69.55  NaN    S  
201       2  CA. 2343  69.55  NaN    S  
324       2  CA. 2343  69.55  NaN    S  
792       2  CA. 2343  69.55  NaN    S  
846       2  CA. 2343  69.55  NaN    S  
863       2  CA. 2343  69.55  NaN    S

dataframe定性信息

In [11]:
import pandas as pd
import numpy as np

text = pd.read_csv('F:/Titanic/train_chinese.csv')
text.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   乘客ID    891 non-null    int64  
 1   是否幸存    891 non-null    int64  
 2   仓位等级    891 non-null    int64  
 3   姓名      891 non-null    object 
 4   性别      891 non-null    object 
 5   年龄      714 non-null    float64
 6   兄弟姐妹个数  891 non-null    int64  
 7   父母子女个数  891 non-null    int64  
 8   船票信息    891 non-null    object 
 9   票价      891 non-null    float64
 10  客舱      204 non-null    object 
 11  登船港口    889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


dataframe定量信息

In [12]:
import pandas as pd
import numpy as np

text = pd.read_csv('F:/Titanic/train_chinese.csv')
text.describe()

'''
count : 样本数据大小
mean : 样本数据的平均值
std : 样本数据的标准差
min : 样本数据的最小值
25% : 样本数据25%的时候的值
50% : 样本数据50%的时候的值
75% : 样本数据75%的时候的值
max : 样本数据的最大值
'''

'\ncount : 样本数据大小\nmean : 样本数据的平均值\nstd : 样本数据的标准差\nmin : 样本数据的最小值\n25% : 样本数据25%的时候的值\n50% : 样本数据50%的时候的值\n75% : 样本数据75%的时候的值\nmax : 样本数据的最大值\n'

In [13]:
'''
看看泰坦尼克号数据集中 票价 这列数据的基本统计数据
'''
text['票价'].describe()

count    891.000000
mean      32.204208
std       49.693429
min        0.000000
25%        7.910400
50%       14.454200
75%       31.000000
max      512.329200
Name: 票价, dtype: float64

从上面数据我们可以看出， 一共有891个票价数据， 平均值约为：32.20， 标准差约为49.69，说明票价波动特别大， 25%的人的票价是低于7.91的，50%的人的票价低于14.45，75%的人的票价低于31.00， 票价最大值约为512.33，最小值为0。